dulwich/repo.py

```diff
@@ -676,18 +677,26 @@ def __init__(self, root):
             raise NotGitRepository(
                 "No git repository was found at %(path)s" % dict(path=root)
             )
+        commondir = self.get_named_file(COMMONDIR)
+        if commondir:
+            for line in commondir:
+                self._commondir = os.path.join(self.controldir(), line.rstrip("\n"))
```

Comment:
@jelmer Would you agree that I add an other argument `mode='rb'` to `get_named_file`?